beacon_node/beacon_chain/src/data_availability_checker/error.rs

```diff
@@ -27,32 +28,76 @@ pub enum Error {
 
 #[derive(PartialEq, Eq)]
 pub enum ErrorCategory {
-    /// Internal Errors (not caused by peers)
-    Internal,
-    /// Errors caused by faulty / malicious peers
-    Malicious,
+    /// Internal Errors (not caused by peers).
+    ///
+    /// An internal non-recoverable error is permanent and block processing should not be
+    /// re-attempted.
+    Internal { recoverable: bool },
```

Comment:
```suggestion
    Internal { retry: bool },
```
In reviewing this, I found that for a lot of errors, it was really ambiguous what "recoverable" meant in the context.

Instead, we could just indicate the action we want sync to take given the error which would reduce the ambiguity imo.

Reply:
Also, thinking more about it, internal + recoverable/retry doesn't make much sense to me as a variant.
Currently, the error variants that have this category are
```
ParentUnknown/ReconstructColumnsError -> needs a special case to handle

FutureSlot -> i think this should be a malicious error and not internal assuming our clock is correct

BeaconChainError/SlotClockError -> cannot act on it so not really recoverable or worth retrying

ExecutionPayloadError::RequestFailed -> can't act on it until EL is back online

ExecutionPayloadError::UnverifiedNonOptimisticCandidate -> won't happen after the merge

MissingBlobs/Custody -> this is a malicious error

KzgCommitmentMismatch -> this never happens because we check the inclusion proof
```

Reply:
Moved MissingBlobs/Custody to malicious. 

Agree, but will look at dropping KzgCommitmentMismatch in a future PR